### ISSUE TITLE
question about KinesisVideoRekognitionIntegrationExampleTest

### ISSUE THREAD

**Issue by RileyJia** [aws/amazon-kinesis-video-streams-parser-library#28]
I am running this testcase, but the result video is the same as the input. It cannot draw box around the faces. I have create a collection and index Bezos's face. There is no other instructions about this test. I don't know where I did wrong.

**Comment by bdhandap** [comment]
@Rulovescoding Generally Rekognition takes some time to return the results for the detected faces causing the video to be delayed. If the detected faces are not returned within the playback duration of the video file then the video frames will be rendered without bounding boxes. Can you please increase the [rekognition timeout](https://github.com/aws/amazon-kinesis-video-streams-parser-library/blob/master/src/test/java/com/amazonaws/kinesisvideo/parser/examples/KinesisVideoRekognitionIntegrationExampleTest.java#L64) to a higher value so that video frames will wait little bit longer until the rekognition results are returned?

Also a better experiment is to test with a live producer (camera) instead a short test file so that the video playback duration is much longer. Please take a look at [KVS producer SDK demo applications](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#sample-demo-applications-for-sending-video-to-kinesis-video-streams) to know how to send live stream to KVS.

**Comment by Rulovescoding** [comment]
I have tried to set the timeout to like 300 and 1000. It would return an
error said about sleeping thread. The video frame viewer would be closed
after the video display for 3 to 5 seconds. That’s definitely not the right
result, I am not sure how big should I set the timeout variable.

**Comment by manabu-tsunokai** [comment]
I have the same problem, no bounding box is drawn.
I have tried to set the timeout 100 to 20000, no success.
I have also tried a live producer using the camera of my macbook air, but the results were the same.
The log shows that processRecords() of KinesisRecordProcessor is not called at all.
The console of Kinesis Data Stream shows some data transfer is going on, but the chart of GetRecords.Bytes shows the values are always 0.
I can successfully get the rekognition results including matched face through awscli using commands like "get-shard-iterator", "get-records".
I would appreciate any help, thank you.

Region AP_NORTHEAST_1
MacOS 10.14.5
JDK 1.8.0_131

**Comment by bdhandap** [comment]
@manabu-tsunokai If GetRecords.Bytes is 0 then I think the RekognitionStreamProcessor is either not started or encountering some error. Can you do a describe on the stream process and check whether the status is RUNNING?

https://docs.aws.amazon.com/rekognition/latest/dg/streaming-video-troubleshooting.html

**Comment by manabu-tsunokai** [comment]
Thanks for the response. I confirmed that the stream processor is running.
But after some trial and error, I solved the problem.
I ran the test with a live producer, setting the argument of example.execute() to very high (such as 300L). Then it it started to show the bounding box 20 or 30 seconds after starting the test, although the frame rate is very very slow.
Maybe the machine power is not sufficient to run the test smoothly. (I use macbook air early 2015)
